### PR TITLE
Refine List of Submissions Report (CAT)

### LINKED ISSUE #6698: "Most Recent" not accurate for QA submissions on Test
On Test: The "Most Recent" information shown in the Submission List is not accurate for QA data submitted. 

Acceptance Test:
- For QA tests, "Most Recent" should show "Yes" for the latest submission of a specific QA file for a specific configuration. E.g., ORIS 3 unit 1 RATA test # 1234. 
- "Most Recent" should show "No" for a specific QA file for a specific configuration when a more recent version of that QA has been submitted, i.e., a resubmission occurs.

Identifying info for QA data:
- ORIS & Location
- Data Type: Test, TEE, Event
- For Tests: test type, test number
- For TEE: Tee type code, yr/quarter, system ID
- For Events: Event code, date/hour, system ID, component ID

### LINKED ISSUE #6705: Add identifying information to Submission Report
We need to repurpose the "Reporting Period" column to provide more identifying information.

Acceptance Tests:
- The "Reporting Period" column is now labeled "Identifying Information" instead.
- For MP submissions, the column is null.
- For emissions submissions, the column includes the reporting period information.
- For QA submissions, the column includes the data type (Test, TEE, Event) as well as the following information:
-- For Tests: test type, test number
-- For TEE: TEE type code, yr/quarter, system ID
-- For Events: Event code, date/hour, system ID, component ID

### LINKED ISSUE #6706: Additional filters for Submission Report
Add the following as new filters to the Submission Report:

- MP Location
- QA Data Type (Test, Event, TEE)
- Test Type (RATA, etc.)

Acceptance Tests:
- MP Location is available as a filter for the report and brings back accurate information based on the search parameters.
- QA Data Type is available as a filter for the report and brings back accurate information based on the search parameters.
- Test Type is available as a filter for the report and brings back accurate information based on the search parameters.

### LINKED ISSUE #6710: Refine List of Submissions Report (CAT)
(continuation of #6622)

Items that need to be fixed:

- Some of the columns are too wide, some too narrow (you can't read the whole name of the column without hovering) and they can't be manually adjusted. The columns should autofit the information and the headers should wrap if needed. Update on 7/10/25: Moving this issue to a separate ticket.
- [x] Remove the checkbox associated with each row. This functionality is OBE.
- [x] Name of column should be "MP Location(s)", not "MP location"
- [x] Time of submission is not being shown. It should be showing both the date and time.
- [x] "Facility ID" column sorts as a string. It should sort as a number.
- [x] "Reporting Period" column does not sort.
- [x] "Submission Type" column does not sort alphabetically.

Acceptance Criteria 

1. In Error Suppression UI make sure that locations work properly and in Submission Report as well.

### PR DESCRIPTION
**Pull request by Jhernandez29**
tickets
https://github.com/US-EPA-CAMD/easey-ui/issues/6710
https://github.com/US-EPA-CAMD/easey-ui/issues/6706
https://github.com/US-EPA-CAMD/easey-ui/issues/6705
https://github.com/US-EPA-CAMD/easey-ui/issues/6698

changes for https://github.com/US-EPA-CAMD/easey-ui/issues/6710: 
The "Critical Error Level 1", "Critical Error Level 2", "Non-Critical", "Informational Message", "Administrative Override" columns After Teams discussion (5/29, Ann, Dwayne, Jonas, Mark), we have decided to not include these columns in the report.

changes for https://github.com/US-EPA-CAMD/easey-ui/issues/6706:
QA Data Type - Test, QCE, or TEE.  Will need to add a new qa_data_type_cd to the view (TEST if test_sum_id is not null,  QCE if qa_cert_Event_id is not null, TEE if test_extenstion_exemption_id is not null, otherwise NULL)
Test Type - need to add test_type_cd column to view.  left outer join test_summary from submission_queue on test_sum_id.

changes for https://github.com/US-EPA-CAMD/easey-ui/issues/6705:
The "Reporting Period" column is now labeled "Identifying Information" instead.
For MP submissions, the column is null.
For emissions submissions, the column includes the reporting period information.
For QA submissions, the column includes the data type (Test, TEE, Event) as well as the following information:
-- For Tests: test type, test number
-- For TEE: TEE type code, yr/quarter, system ID
-- For Events: Event code, date/hour, system ID, component ID

changes for https://github.com/US-EPA-CAMD/easey-ui/issues/6698:
QA test should be 'Yes' for the latest submission but 'No' 
when a more recent version of that QA has been summited 
"resubmission occurs" using  lsq.status_cd = 'COMPLETE' 
we make sure the submission is a "Yes" or a "No"

### CODE REVIEW

File: camdecmps/views/vw_qa_cert_event_maintenance.sql [hunk 1]
```diff
@@ -8,7 +8,7 @@ AS SELECT qce.qa_cert_event_id AS cert_event_id,
     qce.resub_explanation,
     COALESCE(up.oris_code, spp.oris_code) AS oris_code,
     COALESCE(u.unitid, sp.stack_name) AS unit_stack,
-    ms.system_identifier,
+    ms.system_identifier,amdecmps.vw_qa_test_extens_exempt_maintenance
```
Comment (djw4erg) [inline camdecmps/views/vw_qa_cert_event_maintenance.sql]: This looks like a typo, does not match the version currently in the database, and would return an error.

File: camdecmpsaux/views/vw_submission_list.sql [hunk 2]
```diff
@@ -26,69 +63,150 @@ select  fac.oris_code,
        sbq.queued_time,
        sev.severity_cd_description as severity_level,
         (SELECT
```
Comment (djw4erg) [inline camdecmpsaux/views/vw_submission_list.sql]: I don't believe that the select is needed here.  You could use the following.

```
case
    when sbq.submission_id = camdecmpsaux.get_last_submission( sbs.mon_plan_id, sbq.rpt_period_id, sbq.process_cd,sbq.test_sum_id, sbq.qa_cert_event_id, sbq.test_extension_exemption_id )
    then 'Yes'
    else 'No'
end most_recent
```
instead of
```
(SELECT
         case 
         when sbq.submission_id = camdecmpsaux.get_last_submission(sbs.mon_plan_id,sbq.rpt_period_id,sbq.process_cd,sbq.test_sum_id,sbq.qa_cert_event_id,sbq.test_extension_exemption_id) 
         then 'Yes'
         else  'No'
         END 
)  as most_recent
```

File: camdecmpsaux/functions/get_last_submission.sql [hunk 3]
```diff
@@ -1,26 +1,43 @@
...
-    where  sbs.mon_plan_Id = v_mon_plan_id
-    and  sbq.process_cd = v_submission_type
-    and  ( sbq.process_cd != 'EM' or sbq.rpt_period_id = v_rpt_period_id )
-    order
-        by  sbq.queued_time desc
-    limit  1;
+    from  camdecmpsaux.SUBMISSION_QUEUE lsq
+                    join camdecmpsaux.SUBMISSION_SET lss using ( submission_set_id )
+             where  (
+                        ( ( v_process_cd = 'MP' ) and ( lss.mon_plan_id = v_mon_plan_id  ))
```
Comment (djw4erg) [inline camdecmpsaux/functions/get_last_submission.sql:22]: For each of these you should make sure that lsq.process_cd equals v_process_cd.  This is especially true when v_process_cd equals 'MP' because lss.mon_plan_id will equal v_mon_plan_id even when there is not an associated 'MP' SUBMISSION_QUEUE  row.

File: camdecmpsaux/views/vw_submission_list.sql [hunk 4]
```diff
@@ -26,69 +63,150 @@ select  fac.oris_code,
...
-    	(SELECT 
-    	CASE 
-        	WHEN COALESCE(COUNT(cl.chk_session_id), 0) > 0 THEN 'Yes'
-        	ELSE 'No'
-    		END AS severity_administrative_override
-		FROM camdecmpsaux.check_session cs
-		JOIN camdecmpsaux.check_log cl using ( chk_session_id  )
-		WHERE cl.severity_cd IN ('ADMNOVR','FORGIVE')
-    	AND cs.submission_id = sbq.submission_id),
+    (SELECT
```
Comment (djw4erg) [inline camdecmpsaux/views/vw_submission_list.sql]: The Severity subqueries contain two problems:

1. The subqueries find the severities for the **MONITOR_PLAN**s associated with the **PLANT** associated with the **SUBMISSION_ID**.
   1. They should find the severities for the **MP**, **QAT**, **QCE**, **TEE**, or **EM** associated with the **SUBMISSION_ID**.
2. The subqueries return a row for **every existing** **SUBMISSION_QUEUE** row, and then selects the one that matters from the returned rows.

I would suggest using the severity subqueries like those in the following example which handles both problems.

```
select  fac.oris_code,
        fac.facility_name,
        camdecmps.GET_MP_LOCATION_LIST( sbs.mon_plan_id ) AS locations,
        sbq.process_cd,
        case ( sbq.process_cd )
            when 'MP' then sbq.process_cd 
            when 'EM' then sbq.process_cd 
            when 'QA' then
                case
                    when sbq.test_sum_id is not null and sbq.qa_cert_event_id is null and sbq.test_extension_exemption_id is null then 'QAT'
                    when sbq.test_sum_id is null and sbq.qa_cert_event_id is not null and sbq.test_extension_exemption_id is null then 'QCE'
                    when sbq.test_sum_id is null and sbq.qa_cert_event_id is null and sbq.test_extension_exemption_id is not null then 'TEE'
                end
        end as file_type_cd,
        sbq.submission_id,
        -- Critical 1
        case
            when exists
                 (
                    select  1
                      from  camdecmpsaux.CHECK_LOG chl
                     where  chl.chk_session_id = case
                                                    when sbq.process_cd = 'MP' then
                                                    ( select pln.chk_session_id from camdecmps.MONITOR_PLAN pln where pln.mon_plan_id = sbs.mon_plan_id )
                                                    when sbq.process_cd = 'EM' then
                                                    ( select ems.chk_session_id from camdecmps.EMISSION_EVALUATION ems where ems.mon_plan_id = sbs.mon_plan_id and ems.rpt_period_id = sbq.rpt_period_id )
                                                    when sbq.process_cd = 'QA' and sbq.test_sum_id is not null and sbq.qa_cert_event_id is null and sbq.test_extension_exemption_id is null then
                                                    ( select tst.chk_session_id from camdecmps.TEST_SUMMARY tst where tst.test_sum_id = sbq.test_sum_id )
                                                    when sbq.process_cd = 'QA' and sbq.test_sum_id is null and sbq.qa_cert_event_id is not null and sbq.test_extension_exemption_id is null then
                                                    ( select qce.chk_session_id from camdecmps.QA_CERT_EVENT qce where qce.qa_cert_event_id = sbq.qa_cert_event_id )
                                                    when sbq.process_cd = 'QA' and sbq.test_sum_id is null and sbq.qa_cert_event_id is null and sbq.test_extension_exemption_id is not null then
                                                    ( select tee.chk_session_id from camdecmps.TEST_EXTENSION_EXEMPTION tee where tee.test_extension_exemption_id = sbq.test_extension_exemption_id )
                                                 end
                       and  chl.severity_cd = 'CRIT1'
                 )
            then 'Yes'
            else 'No'
        end as severity_critical_1,
        -- Critical 2
        case
            when exists
                 (
                    select  1
                      from  camdecmpsaux.CHECK_LOG chl
                     where  chl.chk_session_id = case
                                                    when sbq.process_cd = 'MP' then
                                                    ( select pln.chk_session_id from camdecmps.MONITOR_PLAN pln where pln.mon_plan_id = sbs.mon_plan_id )
                                                    when sbq.process_cd = 'EM' then
                                                    ( select ems.chk_session_id from camdecmps.EMISSION_EVALUATION ems where ems.mon_plan_id = sbs.mon_plan_id and ems.rpt_period_id = sbq.rpt_period_id )
                                                    when sbq.process_cd = 'QA' and sbq.test_sum_id is not null and sbq.qa_cert_event_id is null and sbq.test_extension_exemption_id is null then
                                                    ( select tst.chk_session_id from camdecmps.TEST_SUMMARY tst where tst.test_sum_id = sbq.test_sum_id )
                                                    when sbq.process_cd = 'QA' and sbq.test_sum_id is null and sbq.qa_cert_event_id is not null and sbq.test_extension_exemption_id is null then
                                                    ( select qce.chk_session_id from camdecmps.QA_CERT_EVENT qce where qce.qa_cert_event_id = sbq.qa_cert_event_id )
                                                    when sbq.process_cd = 'QA' and sbq.test_sum_id is null and sbq.qa_cert_event_id is null and sbq.test_extension_exemption_id is not null then
                                                    ( select tee.chk_session_id from camdecmps.TEST_EXTENSION_EXEMPTION tee where tee.test_extension_exemption_id = sbq.test_extension_exemption_id )
                                                 end
                       and  chl.severity_cd = 'CRIT2'
                 )
            then 'Yes'
            else 'No'
        end as severity_critical_2,
        -- Critical 3
        case
            when exists
                 (
                    select  1
                      from  camdecmpsaux.CHECK_LOG chl
                     where  chl.chk_session_id = case
                                                    when sbq.process_cd = 'MP' then
                                                    ( select pln.chk_session_id from camdecmps.MONITOR_PLAN pln where pln.mon_plan_id = sbs.mon_plan_id )
                                                    when sbq.process_cd = 'EM' then
                                                    ( select ems.chk_session_id from camdecmps.EMISSION_EVALUATION ems where ems.mon_plan_id = sbs.mon_plan_id and ems.rpt_period_id = sbq.rpt_period_id )
                                                    when sbq.process_cd = 'QA' and sbq.test_sum_id is not null and sbq.qa_cert_event_id is null and sbq.test_extension_exemption_id is null then
                                                    ( select tst.chk_session_id from camdecmps.TEST_SUMMARY tst where tst.test_sum_id = sbq.test_sum_id )
                                                    when sbq.process_cd = 'QA' and sbq.test_sum_id is null and sbq.qa_cert_event_id is not null and sbq.test_extension_exemption_id is null then
                                                    ( select qce.chk_session_id from camdecmps.QA_CERT_EVENT qce where qce.qa_cert_event_id = sbq.qa_cert_event_id )
                                                    when sbq.process_cd = 'QA' and sbq.test_sum_id is null and sbq.qa_cert_event_id is null and sbq.test_extension_exemption_id is not null then
                                                    ( select tee.chk_session_id from camdecmps.TEST_EXTENSION_EXEMPTION tee where tee.test_extension_exemption_id = sbq.test_extension_exemption_id )
                                                 end
                       and  chl.severity_cd = 'CRIT3'
                 )
            then 'Yes'
            else 'No'
        end as severity_critical_3,
        -- Non-Critical
        case
            when exists
                 (
                    select  1
                      from  camdecmpsaux.CHECK_LOG chl
                     where  chl.chk_session_id = case
                                                    when sbq.process_cd = 'MP' then
                                                    ( select pln.chk_session_id from camdecmps.MONITOR_PLAN pln where pln.mon_plan_id = sbs.mon_plan_id )
                                                    when sbq.process_cd = 'EM' then
                                                    ( select ems.chk_session_id from camdecmps.EMISSION_EVALUATION ems where ems.mon_plan_id = sbs.mon_plan_id and ems.rpt_period_id = sbq.rpt_period_id )
                                                    when sbq.process_cd = 'QA' and sbq.test_sum_id is not null and sbq.qa_cert_event_id is null and sbq.test_extension_exemption_id is null then
                                                    ( select tst.chk_session_id from camdecmps.TEST_SUMMARY tst where tst.test_sum_id = sbq.test_sum_id )
                                                    when sbq.process_cd = 'QA' and sbq.test_sum_id is null and sbq.qa_cert_event_id is not null and sbq.test_extension_exemption_id is null then
                                                    ( select qce.chk_session_id from camdecmps.QA_CERT_EVENT qce where qce.qa_cert_event_id = sbq.qa_cert_event_id )
                                                    when sbq.process_cd = 'QA' and sbq.test_sum_id is null and sbq.qa_cert_event_id is null and sbq.test_extension_exemption_id is not null then
                                                    ( select tee.chk_session_id from camdecmps.TEST_EXTENSION_EXEMPTION tee where tee.test_extension_exemption_id = sbq.test_extension_exemption_id )
                                                 end
                       and  chl.severity_cd = 'NONCRIT'
                 )
            then 'Yes'
            else 'No'
        end as severity_non_critical,
        -- Informational
        case
            when exists
                 (
                    select  1
                      from  camdecmpsaux.CHECK_LOG chl
                     where  chl.chk_session_id = case
                                                    when sbq.process_cd = 'MP' then
                                                    ( select pln.chk_session_id from camdecmps.MONITOR_PLAN pln where pln.mon_plan_id = sbs.mon_plan_id )
                                                    when sbq.process_cd = 'EM' then
                                                    ( select ems.chk_session_id from camdecmps.EMISSION_EVALUATION ems where ems.mon_plan_id = sbs.mon_plan_id and ems.rpt_period_id = sbq.rpt_period_id )
                                                    when sbq.process_cd = 'QA' and sbq.test_sum_id is not null and sbq.qa_cert_event_id is null and sbq.test_extension_exemption_id is null then
                                                    ( select tst.chk_session_id from camdecmps.TEST_SUMMARY tst where tst.test_sum_id = sbq.test_sum_id )
                                                    when sbq.process_cd = 'QA' and sbq.test_sum_id is null and sbq.qa_cert_event_id is not null and sbq.test_extension_exemption_id is null then
                                                    ( select qce.chk_session_id from camdecmps.QA_CERT_EVENT qce where qce.qa_cert_event_id = sbq.qa_cert_event_id )
                                                    when sbq.process_cd = 'QA' and sbq.test_sum_id is null and sbq.qa_cert_event_id is null and sbq.test_extension_exemption_id is not null then
                                                    ( select tee.chk_session_id from camdecmps.TEST_EXTENSION_EXEMPTION tee where tee.test_extension_exemption_id = sbq.test_extension_exemption_id )
                                                 end
                       and  chl.severity_cd = 'INFORM'
                 )
            then 'Yes'
            else 'No'
        end as severity_informational,
        -- Administrative Override or Forgive
        case
            when exists
                 (
                    select  1
                      from  camdecmpsaux.CHECK_LOG chl
                     where  chl.chk_session_id = case
                                                    when sbq.process_cd = 'MP' then
                                                    ( select pln.chk_session_id from camdecmps.MONITOR_PLAN pln where pln.mon_plan_id = sbs.mon_plan_id )
                                                    when sbq.process_cd = 'EM' then
                                                    ( select ems.chk_session_id from camdecmps.EMISSION_EVALUATION ems where ems.mon_plan_id = sbs.mon_plan_id and ems.rpt_period_id = sbq.rpt_period_id )
                                                    when sbq.process_cd = 'QA' and sbq.test_sum_id is not null and sbq.qa_cert_event_id is null and sbq.test_extension_exemption_id is null then
                                                    ( select tst.chk_session_id from camdecmps.TEST_SUMMARY tst where tst.test_sum_id = sbq.test_sum_id )
                                                    when sbq.process_cd = 'QA' and sbq.test_sum_id is null and sbq.qa_cert_event_id is not null and sbq.test_extension_exemption_id is null then
                                                    ( select qce.chk_session_id from camdecmps.QA_CERT_EVENT qce where qce.qa_cert_event_id = sbq.qa_cert_event_id )
                                                    when sbq.process_cd = 'QA' and sbq.test_sum_id is null and sbq.qa_cert_event_id is null and sbq.test_extension_exemption_id is not null then
                                                    ( select tee.chk_session_id from camdecmps.TEST_EXTENSION_EXEMPTION tee where tee.test_extension_exemption_id = sbq.test_extension_exemption_id )
                                                 end
                       and  chl.severity_cd =  any ( array[ 'ADMNOVR', 'FORGIVE' ] )
                 )
            then 'Yes'
            else 'No'
        end as severity_administrative_override,
        sbs.mon_plan_id,
        case when sbq.process_cd = 'EM' then sbq.rpt_period_id end as rpt_period_id,
        case when sbq.process_cd = 'QA' then sbq.test_sum_id end as test_sum_id,
        case when sbq.process_cd = 'QA' then sbq.qa_cert_event_id end as qa_cert_event_id,
        case when sbq.process_cd = 'QA' then sbq.test_extension_exemption_id end as test_extension_exemption_id
  from  camdecmpsaux.SUBMISSION_QUEUE sbq
        join camdecmpsaux.SUBMISSION_SET sbs using( submission_set_id )
        join camd.PLANT fac using ( fac_id )
        join camdecmps.MONITOR_PLAN pln using ( mon_plan_id )
 order
    by  oris_code,
        locations,
        process_cd,
        submission_id;
```

**Review by djw4erg — CHANGES_REQUESTED** [review]

**Review by djw4erg — APPROVED** [review]
